src/test/java/uk/nhs/hee/tis/trainee/forms/migration/AddPartialDeleteMetadataToS3Test.java

```diff
@@ -0,0 +1,195 @@
+    objectSummary1.setKey("1");
+    objectSummary2 = new S3ObjectSummary();
+    objectSummary2.setKey("2");
+    objectSummary3 = new S3ObjectSummary();
+    objectSummary3.setKey("3");
+
+    objects = new ObjectListing();
+    objects.setTruncated(true);
+    List<S3ObjectSummary> objectSummaries = objects.getObjectSummaries();
+    objectSummaries.add(objectSummary1);
+    objectSummaries.add(objectSummary2);
+
+    objects2 = new ObjectListing();
+    objects2.setTruncated(false);
+    List<S3ObjectSummary> objectSummaries2 = objects2.getObjectSummaries();
+    objectSummaries2.add(objectSummary3);
+
+    // set up S3Objects
+    objectContent1 = new ByteArrayInputStream("{\"id\":\"1\"}".getBytes());
+    objectContent2 = new ByteArrayInputStream("{\"id\":\"2\"}".getBytes());
+    objectContent3 = new ByteArrayInputStream("{\"id\":\"3\"}".getBytes());
+
+    metadata = new ObjectMetadata();
+    metadata.addUserMetadata("metaName1", "mataValue1");
+    metadata.addUserMetadata("metaName2", "mataValue2");
+    metadata.addUserMetadata("metaName3", "mataValue3");
+
+    object1 = new S3Object();
+    object1.setBucketName(BUCKET_NAME);
+    object1.setKey("1");
+    object1.setObjectContent(objectContent1);
+    object1.setObjectMetadata(metadata);
+
+    object2 = new S3Object();
+    object2.setBucketName(BUCKET_NAME);
+    object2.setKey("2");
+    object2.setObjectContent(objectContent2);
+    object2.setObjectMetadata(metadata);
+
+    object3 = new S3Object();
+    object3.setBucketName(BUCKET_NAME);
+    object3.setKey("3");
+    object3.setObjectContent(objectContent3);
+    object3.setObjectMetadata(metadata);
+  }
+
+  @Test
+  void shouldAddMetadataToObject() throws IOException {
+    when(s3.listObjects(BUCKET_NAME)).thenReturn(objects);
+    when(s3.listNextBatchOfObjects(objects)).thenReturn(objects2);
+    when(s3.getObject(BUCKET_NAME, "1")).thenReturn(object1);
+    when(s3.getObject(BUCKET_NAME, "2")).thenReturn(object2);
+    when(s3.getObject(BUCKET_NAME, "3")).thenReturn(object3);
+
+    migration.migrate();
+
+    verify(s3, times(3)).putObject(putRequestCaptor.capture());
+    List<PutObjectRequest> putObjectRequests = putRequestCaptor.getAllValues();
+    assertThat("Unexpected put object request count.",
+        putObjectRequests.size(), CoreMatchers.is(3));
+
+    var assertId = 1;
+    for (var putObjectRequest : putObjectRequests) {
+      assertThat("Unexpected bucket.",
+          putObjectRequest.getBucketName(), is(BUCKET_NAME));
+      assertThat("Unexpected key.",
+          putObjectRequest.getKey(), is(Integer.toString(assertId)));
+      final var resultInputStream = putObjectRequest.getInputStream();
+      ObjectMapper mapper = new ObjectMapper();
+      Map<String, Object> resultJsonMap = mapper.readValue(resultInputStream, Map.class);
+      assertThat("Unexpected input stream.",
+          resultJsonMap.get("id"), is(Integer.toString(assertId)));
+
+      final var resultUserMetadata =
```

Comment:
As with other `var` comments, it's hard to know what this call returns outside of an IDE